flink-optimizer/src/main/java/org/apache/flink/optimizer/plantranslate/JobGraphGenerator.java

```diff
@@ -1138,12 +1138,13 @@ private DistributionPattern connectJobVertices(Channel channel, int inputNumber,
 		
 		if (channel.getShipStrategy() == ShipStrategyType.PARTITION_RANGE) {
 			
-			final DataDistribution dataDistribution = channel.getDataDistribution();
+			DataDistribution dataDistribution = channel.getDataDistribution();
 			if (dataDistribution != null) {
 				sourceConfig.setOutputDataDistribution(dataDistribution, outputIndex);
 			} else {
-				throw new RuntimeException("Range partitioning requires data distribution");
-				// TODO: inject code and configuration for automatic histogram generation
```

Comment:
As the TODO comment says, here should go the sampling and distribution building code :-)